blockchain/chain/src/store/chain_store.rs

```diff
@@ -192,6 +187,34 @@ where
         }
         Ok(())
     }
+
+    pub async fn sub_head_changes(&mut self) -> Result<Receiver<HeadChange>, Error> {
+        let subscribed_head_change = self.subscribe();
+        info!("sub head changes");
+        let head = self
+            .heaviest_tipset()
+            .ok_or_else(|| Error::Other("Could not get heaviest tipset".to_string()))?;
+        let (mut sender, reciever) = channel(16);
+        sender
+            .send(HeadChange::Current(head))
+            .await
+            .map_err(|e| Error::Other(format!("Could not send to channel {:?}", e)))?;
+        task::spawn(async move {
+            let sender = &sender.clone();
+            let wait_for_each = subscribed_head_change.for_each(|s| async move {
+                sender
+                    .clone()
+                    .send(s)
+                    .await
+                    .unwrap_or_else(|e| warn!("failed to send to channel : {:?}", e));
+            });
+            let dur = Duration::from_millis(10);
+            future::timeout(dur, wait_for_each)
+                .await
+                .unwrap_or_else(|_| info!("waited for head_change for 10 miliseconds"));
```

Comment:
Won't this just immediately time out? what is the purpose of this?

Reply:
This is for cases where the pending future doesn't change which is flostreams design. This will timeout when future in the foreach hasn't  been resolved